### ISSUE TITLE
import-error for jsonschema

### ISSUE THREAD

**Issue by dseomn** [google/pytype#318]
From https://travis-ci.org/dseomn/dotfiles/jobs/534215443
```
$ pytype .
Computing dependencies
Analyzing 17 sources with 0 local dependencies
ninja: Entering directory `/home/travis/build/dseomn/dotfiles/.local/lib/python3/site-packages/.pytype'
[4/17] check snipphthalate.plugin
FAILED: /home/travis/build/dseomn/dotfiles/.local/lib/python3/site-packages/.pytype/pyi/snipphthalate/plugin.pyi 
pytype-single --imports_info /home/travis/build/dseomn/dotfiles/.local/lib/python3/site-packages/.pytype/imports/snipphthalate.plugin.imports --module-name snipphthalate.plugin -V 3.5 -o /home/travis/build/dseomn/dotfiles/.local/lib/python3/site-packages/.pytype/pyi/snipphthalate/plugin.pyi --analyze-annotated --nofail --quick /home/travis/build/dseomn/dotfiles/.local/lib/python3/site-packages/snipphthalate/plugin.py
File "/home/travis/build/dseomn/dotfiles/.local/lib/python3/site-packages/snipphthalate/plugin.py", line 19, in <module>: Can't find module 'jsonschema'. [import-error]
File "/home/travis/build/dseomn/dotfiles/.local/lib/python3/site-packages/snipphthalate/plugin.py", line 78, in __init__: No attribute 'exec_module' on _importlib_modulespec._Loader [attribute-error]
  In Union[Any, _importlib_modulespec._Loader]
For more details, see https://google.github.io/pytype/errors.html.
[6/17] check dconf_load_yaml.main
FAILED: /home/travis/build/dseomn/dotfiles/.local/lib/python3/site-packages/.pytype/pyi/dconf_load_yaml/main.pyi 
pytype-single --imports_info /home/travis/build/dseomn/dotfiles/.local/lib/python3/site-packages/.pytype/imports/dconf_load_yaml.main.imports --module-name dconf_load_yaml.main -V 3.5 -o /home/travis/build/dseomn/dotfiles/.local/lib/python3/site-packages/.pytype/pyi/dconf_load_yaml/main.pyi --analyze-annotated --nofail --quick /home/travis/build/dseomn/dotfiles/.local/lib/python3/site-packages/dconf_load_yaml/main.py
File "/home/travis/build/dseomn/dotfiles/.local/lib/python3/site-packages/dconf_load_yaml/main.py", line 44, in <module>: Can't find module 'jsonschema'. [import-error]
For more details, see https://google.github.io/pytype/errors.html#import-error.
ninja: build stopped: subcommand failed.
The command "pytype ." exited with 1.
```

jsonschema is installed, and there are passing unit tests that cover those `import jsonschema` lines.

**Comment by rchen152** [comment]
Huh. I reproduced this issue with Python 3.6 and pytype-2019.05.15 by doing:
```
pip install pytype jsonschema
echo "import jsonschema" > foo.py
pytype foo.py
```
But jsonschema is in the virtualenv's lib/python3.6/site-packages directory just like all the packages we *can* find, and it has a `__init__.py` file.

**Comment by ic** [comment]
The same error occurs for many modules, but not all modules. For example, at time of writing, I get the error with `python-dotenv` and `psycopb2`, but _not_ with `flask` (all in a Virtualenv for Python 3.7).

Perhaps a key difference between these modules can lead to a fix.

**Comment by rchen152** [comment]
Ok, this one was bizarre, but I think I've figured it out. There's a broken import in importlab: https://github.com/google/importlab/blob/92090a0b4421137d1369c2ed952eda6bb4c7a155/importlab/import_finder.py#L15. It needs to be `import importlib.util` instead. Unless I run into anything else strange, I should have this fixed later today.

What's bizarre is that this used to be fine and mysteriously stopped working some time between April 11 and April 16, which I figured out by looking at the commit history of python-fire and seeing that pytype started reporting an import error on code it was previously happy with: https://github.com/google/python-fire/commits/master. ¯\_(ツ)_/¯
